core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -488,7 +510,13 @@ private[serializer] object KryoSerializer {
     classOf[StorageLevel],
     classOf[CompressedMapStatus],
     classOf[HighlyCompressedMapStatus],
+    classOf[BitSet],
     classOf[CompactBuffer[_]],
+    classOf[OpenHashSet[Int]],
```

Comment:
Are these actually different types, if the generic type is all that varies?

Reply:
Like [https://github.com/apache/spark/blob/master/graphx/src/main/scala/org/apache/spark/graphx/GraphXUtils.scala#L45](url), I think they are different, since type specialization is used in [https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/collection/OpenHashSet.scala#L44](url)